File: src/main/java/edu/kit/iti/formal/stvs/util/AsyncTaskCompletedHandler.java
package edu.kit.iti.formal.stvs.util;

/**
 * This interface represents a handler that gets called after a {@link JavaFxAsyncProcessTask} has completed its
 * work.
 */
public interface AsyncTaskCompletedHandler<T> {
  void onSuccess(T computedValue);

  void onException(Exception exception);
}

File: stvs/src/main/kotlin/edu/kit/iti/formal/stvs/util/AsyncTaskCompletedHandler.kt
package edu.kit.iti.formal.stvs.util

/**
 * This interface represents a handler that gets called after a [JavaFxAsyncTask] has
 * completed its. work.
 */
interface AsyncTaskCompletedHandler<T> {
    fun onSuccess(computedValue: T)

    fun onException(exception: Exception?)
}
